Sources/LiveKit/Support/FFTProcessor.swift

```diff
@@ -0,0 +1,188 @@
+        self.scaleType = scaleType
+    }
+
+    func computeBands(minFrequency: Float, maxFrequency: Float, bandsCount: Int, sampleRate: Float) -> FFTComputeBandsResult {
+        let actualMaxFrequency = min(sampleRate.nyquistFrequency, maxFrequency)
+        var bandMagnitudes = [Float](repeating: 0.0, count: bandsCount)
+        var bandFrequencies = [Float](repeating: 0.0, count: bandsCount)
+
+        let magLowerRange = _magnitudeIndex(for: minFrequency, sampleRate: sampleRate)
+        let magUpperRange = _magnitudeIndex(for: actualMaxFrequency, sampleRate: sampleRate)
+        let ratio = Float(magUpperRange - magLowerRange) / Float(bandsCount)
+
+        for i in 0 ..< bandsCount {
+            let magsStartIdx = Int(floorf(Float(i) * ratio)) + magLowerRange
+            let magsEndIdx = Int(floorf(Float(i + 1) * ratio)) + magLowerRange
+
+            let count = magsEndIdx - magsStartIdx
+            if count > 0 {
+                if scaleType == .linear {
+                    // Linear scale averaging
+                    bandMagnitudes[i] = _computeAverage(magnitudes, magsStartIdx, magsEndIdx)
+                }
+            } else {
+                // Single value case
+                bandMagnitudes[i] = magnitudes[magsStartIdx]
+            }
+
+            // Compute average frequency
+            bandFrequencies[i] = _averageFrequencyInRange(magsStartIdx, magsEndIdx, sampleRate: sampleRate)
+        }
+
+        return FFTComputeBandsResult(count: bandsCount, magnitudes: bandMagnitudes, frequencies: bandFrequencies)
+    }
+
+    @inline(__always) private func _magnitudeIndex(for frequency: Float, sampleRate: Float) -> Int {
+        Int(Float(magnitudes.count) * frequency / sampleRate.nyquistFrequency)
+    }
+
+    @inline(__always) private func _computeAverage(_ array: [Float], _ startIdx: Int, _ stopIdx: Int) -> Float {
+        var mean: Float = 0
+        let count = stopIdx - startIdx
+        array.withUnsafeBufferPointer { bufferPtr in
+            let ptr = bufferPtr.baseAddress! + startIdx
+            vDSP_meanv(ptr, 1, &mean, UInt(count))
+        }
+        return mean
+    }
+
+    @inline(__always) private func _computeBandwidth(for sampleRate: Float) -> Float {
+        sampleRate.nyquistFrequency / Float(magnitudes.count)
+    }
+
+    @inline(__always) private func _averageFrequencyInRange(_ startIndex: Int, _ endIndex: Int, sampleRate: Float) -> Float {
+        let bandwidth = _computeBandwidth(for: sampleRate)
+        return (bandwidth * Float(startIndex) + bandwidth * Float(endIndex)) / 2
+    }
+}
+
+class FFTProcessor {
+    public enum WindowType {
+        case none
+        case hanning
+        case hamming
+    }
+
+    public enum ScaleType {
+        case linear
+        case logarithmic
+    }
+
+    public let bufferSize: Int
+    public let windowType: WindowType
+    public let scaleType: ScaleType
+
+    private let bufferHalfSize: Int
+    private let bufferLog2Size: Int
```

Comment:
you should consider declaring these int types as `vDSP_Length` instead. At the very least, they should be `UInt` here and cast to `vDSP_Length` when passed to accelerate later